src/lib/db/segment-store.ts

```diff
@@ -170,6 +170,15 @@ export default class SegmentStore implements ISegmentStore {
         }));
     }
 
+    async getByName(name: string): Promise<ISegment> {
+        const rows: ISegmentRow[] = await this.db
+            .select(this.prefixColumns())
+            .from(T.segments)
+            .where({ name });
+
+        return this.mapRow(rows[0]);
```

Comment:
What happens when rows is empty?

Reply:
If rows is empty, then `mapRow` will throw a `NotFoundError` which is what we are using for our logic.